testnet-contracts/contracts/Oracle.sol

```diff
@@ -0,0 +1,297 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../../node_modules/openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./Recoverer.sol";
+
+contract Oracle is Recoverer {
+
+    using SafeMath for uint256;
+
+    /*
+    * @dev: Public variable declarations
+    */
+    address public provider;
+    uint256 public numbValidators;
+    uint256 public totalPower;
+    uint256 public seqCounter = 0;
+    address[] public validators;
+    mapping(address => bool) public activeValidators;
+    mapping(address => uint256) public powers;
+    // Maps CosmosBridgeClaim id to OracleClaims made on it by validators
+    mapping(uint256 => OracleClaim[]) public oracleClaims;
+    // Maps validator's address to each CosmosBridgeClaim they've made
+    mapping(address => uint256[]) public validatorOracleClaims;
+
+    /*
+    * @dev: Temporary buffer which records address use in a processProphecyClaim attempt,
+    *       since the Solidity compiler does not yet implement memory (local) mappings
+    */
+    mapping(address => bool) internal usedAddresses;
```

Comment:
Using storage is one of the most gas intensive operations. I'm not sure it makes sense to use this when we only need it for memory purposes, and we likely want to consider an alternative data structure/algorithm logic for doing the usedAddresses check even if it means worse-looking code. Can discuss on call